components/org.wso2.carbon.micro.integrator.extensions/org.wso2.carbon.micro.integrator.management.apis/src/main/java/org/wso2/carbon/micro/integrator/management/apis/DataServiceResource.java

```diff
@@ -0,0 +1,158 @@
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.micro.integrator.management.apis;
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import com.google.gson.Gson;
+import org.apache.axis2.AxisFault;
+import org.apache.axis2.description.AxisService;
+import org.apache.axis2.engine.AxisConfiguration;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.synapse.MessageContext;
+import org.apache.synapse.config.SynapseConfiguration;
+import org.apache.synapse.core.axis2.Axis2MessageContext;
+import org.json.JSONObject;
+import org.wso2.carbon.dataservices.common.DBConstants;
+import org.wso2.carbon.dataservices.core.DBUtils;
+import org.wso2.carbon.dataservices.core.description.query.Query;
+import org.wso2.carbon.dataservices.core.engine.DataService;
+import org.wso2.carbon.inbound.endpoint.internal.http.api.APIResource;
+import org.wso2.carbon.micro.integrator.management.apis.models.dataServices.DataServiceInfo;
+import org.wso2.carbon.micro.integrator.management.apis.models.dataServices.DataServiceSummary;
+import org.wso2.carbon.micro.integrator.management.apis.models.dataServices.DataServicesList;
+import org.wso2.carbon.micro.integrator.management.apis.models.dataServices.QuerySummary;
+import org.wso2.carbon.service.mgt.ServiceAdmin;
+import org.wso2.carbon.service.mgt.ServiceMetaData;
+
+public class DataServiceResource extends APIResource {
+    private final Log log = LogFactory.getLog(DataService.class);
+
+    public DataServiceResource(String urlTemplate) {
+        super(urlTemplate);
+    }
+
+    @Override
+    public Set<String> getMethods() {
+        Set<String> methods = new HashSet<>();
+        methods.add("GET");
+        methods.add("POST");
+        return methods;
+    }
+
+    @Override
+    public boolean invoke(MessageContext msgCtx) {
+        buildMessage(msgCtx);
+        String param = Utils.getQueryParameter(msgCtx, "dataServiceName");
+
+        try {
+            if (param != null) {
+                // data-service specified by name
+                populateDataServiceByName(msgCtx, param);
+            } else {
+                // list of all data-services
+                populateDataServiceList(msgCtx);
+            }
+        } catch (AxisFault axisFault) {
+            log.error(axisFault.getStackTrace());
```

Comment:
Isn't better to send an error response to the caller when and AxisFault is thrown?